libraries/AP_Common/tests/test_cpp.cpp

```diff
@@ -0,0 +1,36 @@
+#include <AP_gtest.h>
+#include <AP_HAL/AP_HAL.h>
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
```

Comment:
hal should not be required.  Which is why it was `int` elsewhere

Similarly elsewhere

Reply:
In most case the hal was require because of very deep hal function used. I have removed it were it was possible.